### ISSUE TITLE
Suggestion: Add rule that disallows glob based exports

### ISSUE THREAD

**Issue by Nokel81** [import-js/eslint-plugin-import#2928]
As a style rule, I think that having a rule that lints against `export * from` would be useful.

The following code is valid:
```ts
export { type Foo, Bar } from "./foo";
export type { Bat } from "./foo";
```

The following code is invalid:
```ts
export * from "./foo";
```

**Comment by ljharb** [comment]
You can use no-restricted-syntax for this.

**Comment by Nokel81** [comment]
Ah excellent, sorry for the noise then.

**Comment by Nokel81** [comment]
For anyone else trying to find this, the estree node name is `ExportAllDeclaration`

**Comment by ljharb** [comment]
I'd suggest restricting `import *` in the same way.
